app/controllers/heroku/resources_controller.rb

```diff
@@ -9,6 +9,12 @@ def create
         oauth_grant_code: oauth_grant_code,
       )
 
+      # will be done in a background job once we have jobs set up (very soon)
+      GrantCodeExchanger.new(
+        heroku_uuid: heroku_uuid,
+        oauth_grant_code: oauth_grant_code,
```

Comment:
store `expires_at` for `oauth_grant_code` and when this runs check on whether it is expired or not. If it is expired, throw an error. 

if it is expired, partner has to re-run the request. see: https://github.com/heroku/addons.heroku.com/blob/master/app/jobs/v3/reprice_existing_installs_job.rb